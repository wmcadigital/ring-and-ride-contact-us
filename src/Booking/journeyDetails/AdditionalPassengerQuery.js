import { useEffect } from "react";
import PropTypes from "prop-types";
import { useFormState } from "react-final-form";

import FormSection from "../../common/FormSection";
import RadioGroup from "../../common/RadioGroup";
import Question from "../../common/Question";
import RadioButton from "../../common/RadioButton";
import FieldError from "../../common/FieldError";
import ProgressIndicator from "../../common/ProgressIndicator";
import getSectionPositionInfo from "../getSectionPosition";
import getAboutJourneySectionName from "./getAboutJourneySectionName";

import { required } from "../../common/validation";

const AdditionalPassengerQuery = ({ setAdditionalPassenger }) => {
  const stateApi = useFormState();
  const formValues = stateApi.values;
  const bookingParty = formValues["bookingParty"];

  const error = stateApi.submitFailed
    ? stateApi.errors?.additionalPassenger
    : null;

  const question =
    bookingParty === "mySelf"
      ? "Do you want to bring anyone else with you on the journey?"
      : `Does ${formValues["firstName"]} want to bring anyone else with them on the journey?`;

  useEffect(() => {
    if (formValues.additionalPassenger) {
      setAdditionalPassenger(formValues.additionalPassenger);
    }
  }, [formValues.additionalPassenger]);

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition={getSectionPositionInfo("2", stateApi)}
        sectionName={getAboutJourneySectionName(stateApi)}
      />
      <Question text={question} />
      <RadioGroup error={error}>
        <label className="wmrards-fe-label" htmlFor="additionalPassenger">
          For example, a carer or guardian. They do not need to be a registered
          Ring and Ride scheme user.
        </label>
        <FieldError text={error} />
        <RadioButton
          key={1}
          label="Yes"
          validation={required}
          value="yes"
          fieldName="additionalPassenger"
        />
        <RadioButton
          key={2}
          label="No"
          validation={required}
          value="no"
          fieldName="additionalPassenger"
        />
      </RadioGroup>
    </FormSection>
  );
};

AdditionalPassengerQuery.propTypes = {
  setAdditionalPassenger: PropTypes.func,
};

AdditionalPassengerQuery.defaultProps = {
  setAdditionalPassenger: () => {},
};

export default AdditionalPassengerQuery;
