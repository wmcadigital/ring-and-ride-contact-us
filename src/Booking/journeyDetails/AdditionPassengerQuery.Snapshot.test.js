import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import AdditionalPassengerQuery from "./AdditionalPassengerQuery";

jest.mock("react-final-form");

describe("AdditionalPassengerQuery - snapshots", () => {
  it("individual is filling out form for themselves", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<AdditionalPassengerQuery />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for someone else", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfSomeone",
        firstName: "John",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<AdditionalPassengerQuery />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
        firstName: "John",
      },
      submitFailed: true,
      errors: { additionalPassenger: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<AdditionalPassengerQuery />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
