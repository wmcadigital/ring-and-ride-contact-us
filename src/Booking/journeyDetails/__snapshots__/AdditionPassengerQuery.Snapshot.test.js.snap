// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalPassengerQuery - snapshots individual is filling out form for someone else 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their journey"
    sectionPosition="Section 2 of 4"
  />
  <Question
    text="Does John want to bring anyone else with them on the journey?"
  />
  <RadioGroup
    error={null}
  >
    <label
      className="wmrards-fe-label"
      htmlFor="additionalPassenger"
    >
      For example, a carer or guardian. They do not need to be a registered Ring and Ride scheme user.
    </label>
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="additionalPassenger"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="additionalPassenger"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;

exports[`AdditionalPassengerQuery - snapshots individual is filling out form for themselves 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your journey"
    sectionPosition="Section 2 of 4"
  />
  <Question
    text="Do you want to bring anyone else with you on the journey?"
  />
  <RadioGroup
    error={null}
  >
    <label
      className="wmrards-fe-label"
      htmlFor="additionalPassenger"
    >
      For example, a carer or guardian. They do not need to be a registered Ring and Ride scheme user.
    </label>
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="additionalPassenger"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="additionalPassenger"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;

exports[`AdditionalPassengerQuery - snapshots renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your journey"
    sectionPosition="Section 2 of 4"
  />
  <Question
    text="Do you want to bring anyone else with you on the journey?"
  />
  <RadioGroup
    error="Required"
  >
    <label
      className="wmrards-fe-label"
      htmlFor="additionalPassenger"
    >
      For example, a carer or guardian. They do not need to be a registered Ring and Ride scheme user.
    </label>
    <FieldError
      text="Required"
    />
    <RadioButton
      fieldName="additionalPassenger"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="additionalPassenger"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;
